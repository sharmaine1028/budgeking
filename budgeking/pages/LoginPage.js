import React, { useState } from "react";
import {
  StyleSheet,
  View,
  StatusBar,
  Image,
  ActivityIndicator,
} from "react-native";
import { auth, db } from "../config/firebase";
import { BlackButton, BrownTextInput } from "../config/reusableButton";
import { Footer } from "../config/reusableText";

export default class LoginPage extends React.Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      isLoading: false,
    };
  }

  updateInputVal = (val, prop) => {
    const state = this.state;
    state[prop] = val;
    this.setState(state);
  };

  onFooterLinkPress = () => {
    this.props.navigation.navigate("Signup");
  };

  handleLogin = () => {
    if (this.state.email === "" || this.state.password === "") {
      alert("Enter details to log in!");
    } else {
      this.setState({
        isLoading: true,
      });
      auth
        .signInWithEmailAndPassword(this.state.email, this.state.password)
        .then((res) => {
          console.log("User logged in successfully!");
          this.setState({
            isLoading: false,
            email: "",
            password: "",
          });
          this.props.navigation.navigate("Home");
        })
        .catch((error) => {
          if (error.code === "auth/invalid-email") {
            alert("Invalid email");
          }

          if (error.code === "auth/user-not-found") {
            alert("User not found.");
          }

          if (error.code === "auth/wrong-password") {
            alert("Wrong password");
          }

          this.setState({
            isLoading: false,
            email: "",
            password: "",
          });

          console.log(error.message);
          this.setState({ isLoading: false });
          this.props.navigation.navigate("Login");
        });
    }
  };

  render() {
    if (this.state.isLoading) {
      return (
        <View style={styles.container}>
          <ActivityIndicator size="large" color="#9E9E9E" alignItems="center" />
        </View>
      );
    }
    return (
      <View style={styles.container}>
        <Image
          style={styles.logo}
          source={require("../assets/crown.png")}
          resizeMethod={"resize"}
        />
        <View style={styles.buttonContainer}>
          <BrownTextInput
            placeholder={"Email"}
            onChangeText={(val) => this.updateInputVal(val, "email")}
            value={this.state.email}
          />
          <BrownTextInput
            placeholder={"Password"}
            onChangeText={(val) => this.updateInputVal(val, "password")}
            value={this.state.password}
            maxLength={15}
            secureTextEntry={true}
          />
          <BlackButton text="Login" onPress={this.handleLogin} />
          <Footer
            desc={"Don't have an account yet?"}
            text={"Sign up"}
            onPress={this.onFooterLinkPress}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  buttonContainer: { top: 50 },
  container: {
    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
    display: "flex",
    flexDirection: "column",
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  logo: {
    bottom: 20,
    width: 223,
    height: 146,
    overflow: "visible",
    resizeMode: "contain",
  },
});
