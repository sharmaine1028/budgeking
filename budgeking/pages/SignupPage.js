import React from "react";
import {
  View,
  StyleSheet,
  StatusBar,
  ActivityIndicator,
  Image,
} from "react-native";
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import colours from "../config/colours";
import { BlackButton, AddButton } from "../config/reusableButton";
import { Footer, BrownTextInput } from "../config/reusableText";
import { auth, db } from "../config/firebase";
import * as ImagePicker from "expo-image-picker";

export default class SignupPage extends React.Component {
  constructor() {
    super();
    this.state = {
      firstName: "",
      email: "",
      password: "",
      imageSource: "",
      uploading: false,
      isLoading: false,
      cameraPermission: false,
    };
  }

  // async componentDidMount() {
  //   if (Platform.OS !== "web") {
  //     const { status } = await ImagePicker.getMediaLibraryPermissionsAsync();
  //     console.log(status);
  //     if (status !== "granted") {
  //       alert("Sorry, we need camera roll permissions to make this work!");
  //     }
  //   }
  // }

  render() {
    if (this.state.isLoading) {
      return (
        <View style={(styles.container, styles.center)}>
          <ActivityIndicator size="large" color="#9E9E9E" />
        </View>
      );
    }
    return (
      <KeyboardAwareScrollView
        style={styles.container}
        scrollEnabled={false}
        contentContainerStyle={{
          alignItems: "center",
          justifyContent: "center",
          flex: 1,
        }}
      >
        <View>
          {this.maybeRenderImage()}
          <View style={styles.button}>
            <AddButton onPress={() => this.addImageButton()} />
          </View>
        </View>

        <BrownTextInput
          placeholder={"Email"}
          onChangeText={(val) => this.updateInputVal(val, "email")}
          value={this.state.email}
        />
        <BrownTextInput
          placeholder={"First name"}
          onChangeText={(val) => this.updateInputVal(val, "firstName")}
          value={this.state.username}
        />
        <BrownTextInput
          placeholder={"Password"}
          onChangeText={(val) => this.updateInputVal(val, "password")}
          value={this.state.password}
          // maxLength={15}
          secureTextEntry={true}
        />

        <BlackButton text="Sign up" onPress={this.handleSignUp} />
        <Footer
          desc={"Already have an account?"}
          text={"Log in"}
          onPress={this.onFooterLinkPress}
        />
      </KeyboardAwareScrollView>
    );
  }

  updateInputVal = (val, prop) => {
    const state = this.state;
    state[prop] = val;
    this.setState(state);
  };

  /**
   * Navigate to Login page
   */
  onFooterLinkPress = () => {
    this.props.navigation.navigate("Login");
  };

  /**
   * Renders profile picture depending on whether user added one.
   * Otherwise, add default profile picture.
   *
   * @returns profile picture that users add, default picture otherwise
   */
  maybeRenderImage = () => {
    if (this.state.imageSource === "") {
      return (
        <Image
          style={styles.image}
          source={require("../assets/loginsignup/profile.png")}
        />
      );
    }

    return (
      <Image style={styles.image} source={{ uri: this.state.imageSource }} />
    );
  };

  /**
   * Calls pick Image function
   */
  addImageButton = async () => {
    this.pickImage();
  };

  /**
   * Allows user to pick image from gallery and updates state accordingly
   */
  pickImage = async () => {
    let pickerResult = await ImagePicker.launchImageLibraryAsync({
      allowsEditing: true,
      aspect: [1, 1],
    }).catch((err) => console.log(error));

    if (!pickerResult.cancelled) {
      this.setState({ imageSource: pickerResult.uri });
    }
  };

  /**
   * Handles errors in filling up of fields.
   * Handles logic of updating firebase authentication and details in firestore
   */
  handleSignUp = () => {
    // If any of the necessary fields are empty, alert user
    if (
      this.state.email === "" ||
      this.state.password === "" ||
      this.state.firstName === ""
    ) {
      alert("Enter details to sign up!");
    } else {
      this.setState({
        isLoading: true,
      });

      // Create new account for user
      auth
        .createUserWithEmailAndPassword(this.state.email, this.state.password)
        .then((res) => {
          // Add in user details in firestore
          db.collection("users").doc(res.user.uid).set({
            name: this.state.firstName,
            email: this.state.email,
            budgetValue: 0.0,
            dateTo: new Date(),
            dateFrom: new Date(),
            timeUserWants: "This Month",
          });

          email = this.state.email;
          db.doc("userLookup").set({
            email: res.user.uid,
          });

          // Update details in firebase authentication
          res.user.updateProfile({
            displayName: this.state.firstName,
            photoURL: this.state.imageSource,
          });

          // Alerts user to log in with new account
          alert("Log in with your new account");

          // Reset state on sign up page
          this.setState({
            isLoading: false,
            username: "",
            email: "",
            password: "",
          });

          // Navigates to login page for user to log in with new account
          this.props.navigation.navigate("Login");
        })
        .catch((error) => {
          if (error.code === "auth/email-already-in-use") {
            alert("That email address is already in use!");
          } else if (error.code === "auth/invalid-email") {
            alert("Invalid email");
          } else {
            alert(error.message);
          }
          console.log(error.message);
          this.setState({ isLoading: false });
          this.props.navigation.navigate("Signup");
        });
    }
  };
}

const styles = StyleSheet.create({
  button: {
    width: 30,
    height: 30,
    borderRadius: 999,
    backgroundColor: colours.lightBrown,
    position: "absolute",
    left: 160,
    top: 160,
  },
  container: {
    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
  },
  center: {
    display: "flex",
    flexDirection: "column",
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  image: {
    width: 200,
    height: 200,
    borderRadius: 999,
  },
});
