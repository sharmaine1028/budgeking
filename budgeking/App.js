import * as React from "react";
import { useState } from "react";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import {
  Platform,
  StyleSheet,
  StatusBar,
  LogBox,
  Modal,
  View,
  Text,
} from "react-native";
import colours from "./config/colours";
import LoginPage from "./pages/LoginPage";
import SignupPage from "./pages/SignupPage";
import HomePage from "./pages/Home/HomePage";
import SettingsPage from "./pages/SettingsPage";
import GoalsPage from "./pages/Goal/GoalsPage";
import BudgetPage from "./pages/Budget/BudgetPage";
import ReportsPage from "./pages/Report/ReportsPage";
import LocationSearch from "./pages/Budget/LocationSearch";
import NewGoal from "./pages/Goal/NewGoal";
import SaveToGoal from "./pages/Goal/SaveToGoal";
import EditGoal from "./pages/Goal/EditGoal";
import AllTableView from "./pages/Home/AllTableView";
import ReportsPagePieChart from "./pages/Report/ReportsPagePieChart";
import ChooseCustomDate from "./pages/Report/ChooseCustomDate";
import ReportsPageTable from "./pages/Report/ReportsPageTable";
import GoalHistory from "./pages/Goal/GoalHistory";
import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { TouchableOpacity } from "react-native-gesture-handler";
import { Header } from "./config/reusableText";
import { BlackButton } from "./config/reusableButton";

const Stack = createStackNavigator();
const Tab = createBottomTabNavigator();

LogBox.ignoreAllLogs();

function MyStack() {
  return (
    <Stack.Navigator
      initialRouteName="Login"
      screenOptions={{ headerShown: false }}
    >
      <Stack.Screen name="Login" component={LoginPage} />
      <Stack.Screen name="Signup" component={SignupPage} />
      <Stack.Screen name="HomePage" component={MyTabs} />
    </Stack.Navigator>
  );
}

function GoalsAll() {
  return (
    <Stack.Navigator
      initialRouteName="Goals"
      screenOptions={{
        headerStyle: {
          backgroundColor: colours.whiteRock,
        },
        cardStyle: { backgroundColor: colours.white },
      }}
    >
      <Stack.Screen
        name="Goals"
        component={GoalsPage}
        options={{
          headerLeft: false,
          headerRight: () =>
            Help(
              "1. Add new goals to achieve. Goals are categorised into short-term and long-term.\n\n 2. Once a goal is added, you can edit and delete your goal or add to your savings to complete the goal by clicking on the 3-dotted icon (...), which will then take you to separate pages to fulfill your request. \n\n3. Blue progress bar shows how far you are from your target savings amount.\n\n4. Click on the dropdown arrow to view more details of your goal. \n\n5. Goal box will turn red if your goal deadline has passed. \n\n6. Show goal history shows past goals you have completed."
            ),
        }}
      />
      <Stack.Screen
        name="New Goal"
        component={NewGoal}
        options={{
          headerRight: () =>
            Help(
              "1. Add a goal description and target savings amount. \n\n2. Click on 'Select' dropdown and choose how often you want to save. Then, choose the goal deadline. Frequency amount (amount you have to save) will be automatically displayed beside '$' sign. \n\n4. You can write any notes regarding your goal. \n\n5. If you want to share your goal with someone else, choose 'Yes'. Then, enter the email of user you want to share with. (Note: The user must have signed up on budgeking as well.)\n\n6. Click 'Add' to add the goal and 'Cancel' to go back to Goals page."
            ),
        }}
      />
      <Stack.Screen
        name="Goal History"
        component={GoalHistory}
        options={{
          headerRight: () =>
            Help(
              "Goal history shows an archive all of your past completed goals. For each goal, you can click on dropdown arrow to view the goal's details."
            ),
        }}
      />
      <Stack.Screen
        name="Edit Goal"
        component={EditGoal}
        options={{
          headerRight: () =>
            Help(
              "Add on, remove or edit any details of the goal you have created."
            ),
        }}
      />
      <Stack.Screen
        name="Save to Goal"
        component={SaveToGoal}
        options={{
          headerRight: () =>
            Help(
              "1. 'Target' shows the goal's target savings amount.\n\n2. 'Current Savings' shows how much you have saved towards your goal. \n\n3. Under, 'Add to Savings', key in the amount of money you want to save towards completion of your goal. Amount cannot be more than what is left towards your target amount."
            ),
        }}
      />
    </Stack.Navigator>
  );
}

function BudgetAll() {
  return (
    <Stack.Navigator
      initialRouteName="Budget"
      screenOptions={{
        headerStyle: {
          backgroundColor: colours.whiteRock,
        },
        cardStyle: { backgroundColor: colours.white },
      }}
    >
      <Stack.Screen
        name="Budget"
        component={BudgetPage}
        options={{
          headerLeft: false,
          headerRight: () =>
            Help(
              "1. Toggle between 'Expense' and 'Income' button to input in data for your expenses or income.\n\n2. Input value spent, date and time, category and notes (optional).\n\n3. Attach a photo of your receipt and add a location by clicking on the camera icon and location icon repectively. (optional) \n\n4. Click 'Add' to add to budget. You can see details of what you added in Home Page under 'Last records' for a quick glance. \n\n5. Click on 'Cancel' to remove all details keyed in currently and start over."
            ),
        }}
      />
      <Stack.Screen
        name="Location Search"
        component={LocationSearch}
        options={{
          headerRight: () =>
            Help(
              "1. In the 'Search' bar, search for your desired location by inputting in the name of your desired location. Or, you can click on 'Current location' to get your approximate current location. \n\n2. Choose the option that is most fitting to the location you want. \n\n3. Click on the tick button to confirm your location and go back to Budget Page."
            ),
        }}
      />
    </Stack.Navigator>
  );
}

function ReportsPageAll() {
  return (
    <Stack.Navigator
      initialRouteName="Report"
      screenOptions={{
        headerStyle: {
          backgroundColor: colours.whiteRock,
        },
        cardStyle: { backgroundColor: colours.white },
      }}
    >
      <Stack.Screen
        name="Report"
        component={ReportsPage}
        options={{
          headerLeft: false,
          headerRight: () =>
            Help(
              "1. Overall balance shows the total savings you have. It is a difference between your all-time total expenses and income.\n\n2. Total spending shows your all-time total expenses. Upon clicking on 'TOTAL SPENDING' box, you will be taken to a 'Pie Chart View' of all your expenses that can be filtered by different time periods."
            ),
        }}
      />
      <Stack.Screen
        name="Pie Chart View"
        component={ReportsPagePieChart}
        options={{
          headerRight: () =>
            Help(
              "1. Scroll down to see the dropdown box. Then, select the time period to filter your expenses by. \n\n2. Pie chart will change to show expenses broken down by category according to the time period you selected. Read the legends for the categories and amount of spent for each category. \n\n 3. 'Show more' button navigates to 'Table View' where you can see more detailed logs of expenses and income filtered by the current time period you have selected."
            ),
        }}
      />
      <Stack.Screen
        name="Custom Date"
        component={ChooseCustomDate}
        options={{
          headerRight: () =>
            Help(
              "1. Click on 'Choose custom date' to choose custom dates for the time period to filter expenses and income by. \n\n2. Select a start date and an end date on the calendar for the custom time period. \n\n3. Click on 'Save' button. \n\n4. Check the dates chosen are correct and click on 'Confirm' button to confirm your choice and go back to 'Pie Chart View'."
            ),
        }}
      />
      <Stack.Screen
        name="Table View"
        component={ReportsPageTable}
        options={{
          headerRight: () =>
            Help(
              "All records are sorted in chronological order. You can toggle between red 'Expenses' button and green 'Income' button to view details of all your expenses and income of the time period you have filtered this by."
            ),
        }}
      />
    </Stack.Navigator>
  );
}

function HomePageAll() {
  return (
    <Stack.Navigator
      initialRouteName="Home"
      screenOptions={{
        headerStyle: {
          backgroundColor: colours.whiteRock,
        },
        cardStyle: { backgroundColor: colours.white },
      }}
    >
      <Stack.Screen
        name="Home"
        component={HomePage}
        options={{
          headerLeft: false,
          headerRight: () =>
            Help(
              "1. Choose monthly/ daily mode by clicking on the pink dropdown.\n\n2. Fill in your allowable budget in the currency textbox (beside blue progress bar) for the month or day depending on the mode you have chosen.\n\n3. Blue progress bar shows how much money you can still spend before you reach your allowable budget limit.\n\n4. Pie chart shows a breakdown of all expenses by categories for the month or day depending on the mode you have chosen.\n\n5. Last records show a more detailed view of all expenses. Upon clicking 'Show more', you will be taken to a page with a detailed view of all your records.\n\n6. Notifications will also appear on the Home Page. Upon clicking on the notification, you will be taken to the page requiring your attention."
            ),
        }}
      />
      <Stack.Screen
        name="All Table View"
        component={AllTableView}
        options={{
          headerRight: () =>
            Help(
              "All records are sorted in chronological order. You can toggle between red 'Expenses' button and green 'Income' button to view details of all your expenses and income of all time."
            ),
        }}
      />
    </Stack.Navigator>
  );
}

const Help = (text) => {
  const [modalVisible, setModalVisible] = useState(false);
  return (
    <View style={styles.centeredView}>
      <Modal
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <View style={styles.modalView}>
          <View style={styles.modal}>
            <Header
              text={"\nHelp"}
              style={{
                fontSize: 18,
                fontWeight: "500",
              }}
            />
            <Text style={styles.modalText}>{text}</Text>
            <BlackButton
              text={"Okay"}
              onPress={() => setModalVisible(!modalVisible)}
            />
          </View>
        </View>
      </Modal>
      <TouchableOpacity onPress={() => setModalVisible(true)}>
        <MaterialCommunityIcons
          name="help-circle-outline"
          size={24}
          color="black"
        />
      </TouchableOpacity>
    </View>
  );
};

const screenOptions = (route, color) => {
  let iconName;
  switch (route.name) {
    case "Home":
      iconName = "home";

      break;
    case "Goal":
      iconName = "piggy-bank-outline";
      break;

    case "Budget":
      iconName = "plus-circle";
      break;

    case "Report":
      iconName = "file-document-outline";
      break;
    case "Settings":
      iconName = "cog";
      break;
    default:
      break;
  }
  return <MaterialCommunityIcons name={iconName} color={color} size={28} />;
};

function MyTabs() {
  return (
    <Tab.Navigator
      screenOptions={({ route }) => ({
        headerStyle: {
          backgroundColor: colours.whiteRock,
        },
        unmountOnBlur: true,
        tabBarActiveTintColor: colours.tomato,
        tabBarStyle: {
          backgroundColor: colours.whiteRock,
          height: 60,
          paddingBottom: 2,
        },
        tabBarHideOnKeyboard: true,
        tabBarIcon: ({ color }) => screenOptions(route, color),
      })}
      sceneContainerStyle={{ backgroundColor: colours.white }}
    >
      <Tab.Screen
        name="Home"
        component={HomePageAll}
        options={{
          headerShown: false,
        }}
      />
      <Tab.Screen
        name="Goal"
        component={GoalsAll}
        options={{
          tabBarHideOnKeyboard: true,
          headerShown: false,
        }}
      />
      <Tab.Screen
        name="Budget"
        component={BudgetAll}
        options={{
          tabBarHideOnKeyboard: true,
          headerShown: false,
        }}
      />
      <Tab.Screen
        name="Report"
        component={ReportsPageAll}
        options={{
          headerShown: false,
        }}
      />

      <Tab.Screen
        name="Settings"
        component={SettingsPage}
        options={{
          headerRight: () =>
            Help(
              "1. Change your profile picture by tapping on the current profile picture and choosing a new picture.\n\n2. Click on edit button beside your current username. Now, you can key in the new username in the textbox. Click 'Change' button to update your username.\n\n3. Click on edit button beside the asterisks. Key in your current password in the pop-up window to verify it is you. Click on 'Verify' button. Now, your current password is unmasked and you can key in the new password in the textbox. Click 'Change' button to update your password.\n\n4. 'Log out' button logs you out of your account and takes you back to Login page."
            ),
        }}
      />
    </Tab.Navigator>
  );
}

export default function App() {
  return (
    <NavigationContainer theme={{ colors: colours.black }}>
      <MyStack />
    </NavigationContainer>
  );
}

// export default function App() {
//   return (
//     <NavigationContainer theme={{ colors: colours.black }}>
//       <MyTabs />
//     </NavigationContainer>
//   );
// }

const styles = StyleSheet.create({
  bottomTabIcon: {
    width: 30,
    height: 30,
    overflow: "visible",
    resizeMode: "contain",
  },
  container: {
    flex: 1,
    backgroundColor: colours.brown,
    alignItems: "center",
    justifyContent: "center",
    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
  },
  centeredView: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    marginRight: 10,
  },
  modalView: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    margin: 10,
  },
  modal: {
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    borderWidth: 1,
    borderColor: "#000",
  },
  modalText: {
    fontSize: 15,
    marginBottom: 10,
    fontWeight: "300",
    textAlign: "justify",
  },
});
