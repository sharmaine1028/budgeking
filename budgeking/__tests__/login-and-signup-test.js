import * as React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { render, screen, fireEvent } from "@testing-library/react-native";

import LoginPage from "../pages/LoginPage";
import SignupPage from "../pages/SignupPage";

import { MyStack } from "../App";

describe("Testing react navigation in login and sign up page", () => {
  test("Login page renders correctly", async () => {
    const component = (
      <NavigationContainer>
        <MyStack />
      </NavigationContainer>
    );

    render(component);

    const loginButton = await screen.findByText("Login");

    expect(loginButton).toBeTruthy();
  });

  test("Clicking on sign up link takes you to sign up page", async () => {
    const component = (
      <NavigationContainer>
        <MyStack />
      </NavigationContainer>
    );

    render(component);

    const loginButton = await screen.findByText("Login");

    fireEvent.press(loginButton);

    const signUpButton = await screen.findByText("Sign up");

    expect(signUpButton).toBeTruthy();
  });
});
