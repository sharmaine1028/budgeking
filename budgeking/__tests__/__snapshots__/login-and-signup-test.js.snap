// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react navigation in login and sign up page Clicking on sign up link takes you to sign up page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": undefined,
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": undefined,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "display": "flex",
                            "flex": 1,
                            "flexDirection": "column",
                            "justifyContent": "center",
                            "paddingTop": 0,
                          }
                        }
                      >
                        <Image
                          resizeMethod="resize"
                          source={1}
                          style={
                            Object {
                              "bottom": 20,
                              "height": 146,
                              "overflow": "visible",
                              "resizeMode": "contain",
                              "width": 223,
                            }
                          }
                        />
                        <View
                          style={
                            Object {
                              "top": 50,
                            }
                          }
                        >
                          <TextInput
                            autoCapitalize="none"
                            onChangeText={[Function]}
                            placeholder="Email"
                            style={
                              Object {
                                "backgroundColor": "#e3dad5",
                                "borderRadius": 25,
                                "borderWidth": 1,
                                "height": 46,
                                "margin": 12,
                                "padding": 10,
                                "width": 320,
                              }
                            }
                            value=""
                          />
                          <TextInput
                            autoCapitalize="none"
                            maxLength={15}
                            onChangeText={[Function]}
                            placeholder="Password"
                            secureTextEntry={true}
                            style={
                              Object {
                                "backgroundColor": "#e3dad5",
                                "borderRadius": 25,
                                "borderWidth": 1,
                                "height": 46,
                                "margin": 12,
                                "padding": 10,
                                "width": 320,
                              }
                            }
                            value=""
                          />
                          <TouchableOpacity
                            activeOpacity={0.7}
                            onPress={[Function]}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "center",
                                  "backgroundColor": "#413E3E",
                                  "borderRadius": 999,
                                  "height": 40,
                                  "justifyContent": "center",
                                  "margin": 2,
                                  "width": 130,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#F5F5F5",
                                    "margin": 1,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Login
                            </Text>
                          </TouchableOpacity>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "marginTop": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#2e2e2d",
                                  "fontSize": 16,
                                }
                              }
                            >
                              Don't have an account yet?
                               
                              <Text
                                onPress={[Function]}
                                style={
                                  Object {
                                    "color": "#788eec",
                                    "fontSize": 16,
                                    "fontWeight": "bold",
                                  }
                                }
                              >
                                Sign up
                              </Text>
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
